File: metnuminputfix.py
import math

print("-"*25 + "NEWTON RAPHSON METHOD" + "-"*25)

print("Masukkan nilai x0 : ", end=' ')
x0 = input()
raem = 100.0
i = 0

while raem>=1 :
    print("-" * 35)
    print("Iterasi i=",i)
    i = i + 1
    print("x0      :", x0)
    f = (1.5 ** float(x0)) - 5
    print("f(x)    :", f)
    fd1 = (1.5 ** float(x0)) * 0.4054651081
    print("f'(x)   :", fd1)
    x1 = float(x0) - (f/fd1)
    print("x1      :", x1)
    rae = ((x1-float(x0))/x1) * 100
    if rae<0:
        raem = rae * (-1)
    else:
        raem = rae
    print("rae     :", raem)
    x0 = x1

print(" ")
print(" ")

print("-"*25 + "SECANT METHOD" + "-"*25)

print("Masukkan nilai x0 : ", end=' ')
X0 = input()
print("Masukkan nilai x1 : ", end=' ')
X1 = input()
raem = 100.0
i = 0

while raem>=1:
    print("-" * 35)
    print("Iterasi i=",i)
    i = i + 1
    print("x0      :", X0)
    print("x1      :", X1)
    fx0 = (1.5 ** float(X0)) - 5
    print("f(x)    :", fx0)
    fx1 = (1.5 ** float(X1)) - 5
    print("f'(x)   :", fx1)
    X2 = float(X1) - ((fx1)*(float(X1)-float(X0))/(fx1-fx0))
    print("x2      :", X2)
    rae = ((X2-float(X1))/X2)*100
    if rae<0:
        raem = rae * (-1)
    else:
        raem = rae
    print("rae     :", raem)
    X0 = X1
    X1 = X2

File: metnuminputfix.c
//program metode numeris newton raphson method dan secant method
#include <stdio.h>
#include <math.h>

void newtonraphson();
void secant();

int main(){

    printf("\n");
    newtonraphson();
    printf("\n");
    secant();

    return 0;

}

void newtonraphson(){

    float x, fx;
    float fd1, xa, rae, raem=100;

    printf("-------------------------NEWTON RAPHSON METHOD-------------------------");
    printf("\n\n");
    printf("Masukkan nilai x0 metode Newton Raphson :");
    scanf("%f", &x);
    printf("\n");

    for(int i=0; raem>=1; i++){ // coding metode newton raphson start

        printf("Iterasi i=%d ------------------------------------------\n", i);

        printf("x0       : %f \n", x);

        fx = pow((1.5),x)-5; //rumus utama
        printf("f(x)     : %f \n", fx);

        fd1 = pow((1.5),x) * 0.4054651081; // <<< mulai algoritma newton rhapson
        printf("f'(x)    : %f \n", fd1);

        xa = x-(fx/fd1);
        printf("x1       : %f \n", xa);

        rae = ((xa-x)/xa)*100;
        if(rae<0){
            raem = rae*(-1);
        }
        else{
            raem = rae;
        }
        
        printf("rae      : %f \n", raem);

        x = xa;

    }

    printf("\n");

}

void secant(){

    float x0, x1, fx0, fx1;
    float x2, rae, raem=100;

    printf("-------------------------SECANT METHOD-------------------------");
    printf("\n\n");
    printf("Masukkan nilai x0 metode Secant :");
    scanf("%f", &x0);
    printf("Masukkan nilai x1 metode Secant :");
    scanf("%f", &x1);
    printf("\n");

    for(int i=0; raem>=1; i++){ // coding metode secant start

        printf("Iterasi i=%d ------------------------------------------\n", i);

        printf("x0       : %f \n", x0);

        printf("x1       : %f \n", x1);

        fx0 = pow((1.5),x0)-5; //rumus utama dimasukin x0
        printf("f(x0)    : %f \n", fx0);

        fx1 = pow((1.5),x1)-5; //rumus utama dimasukin x1
        printf("f(x1)    : %f \n", fx1);

        x2 = x1 - ((fx1 * (x1-x0))/(fx1 - fx0)); // <<< mulai algoritma secant
        printf("x2       : %f \n", x2);

        rae = ((x2-x1)/x2)*100;
        if(rae<0){
            raem = rae*(-1);
        }
        else{
            raem = rae;
        }

        printf("rae      : %f \n", raem);\

        x0 = x1;

        x1 = x2;

    }

}